pandas/core/indexes/category.py

```diff
@@ -419,7 +419,11 @@ def reindex(self, target, method=None, level=None, limit=None,
             raise ValueError("cannot reindex with a non-unique indexer")
 
         indexer, missing = self.get_indexer_non_unique(np.array(target))
-        new_target = self.take(indexer)
+
+        if len(self.codes):
```

Comment:
In the case the target is [1] say and the categorical index is empty then indexer here will be [-1] so not empty.  The error occurs when try to run take on the empty underlying index.